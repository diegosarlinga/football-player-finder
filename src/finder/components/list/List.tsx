import React from 'react';
import './List.scss';
import { FootballPlayer } from '../../store/finder-state';

interface ListProps {
  footballPlayers: Array<FootballPlayer>;
}
const ListHeader: React.FC = () => {
  return (
    <thead>
      <tr>
        <th scope="col">Player</th>
        <th scope="col">Position</th>
        <th scope="col" className="d-none d-sm-table-cell">Nationality</th>
        <th scope="col">Age</th>
      </tr>
    </thead>
  );
}

const ListRow: React.FC<FootballPlayer> = (props: FootballPlayer) => {
  return (
    <tr>
      <td key="1">
        {props.name}
        <br className="d-sm-none"></br>
        <span className="d-sm-none text-secondary">{`(${props.nationality})`}</span>
      </td>
      <td key="2">{props.position}</td>
      <td key="3" className="d-none d-sm-table-cell">{props.nationality}</td>
      <td key="4">{props.age}</td>
    </tr>
  );
}

const ListBody: React.FC<{footballPlayers: Array<FootballPlayer>}> = ({ footballPlayers }) => {
  return (
    <tbody>
      { footballPlayers.map((player, ix) => <ListRow {...player} key={ix}></ListRow>) }
    </tbody>
  );
}

const ListEmpty: React.FC = () => {
  return (
    <caption className="text-center">{'No data found for the given filters'}</caption>
  );
}

const List: React.FC<ListProps> = (props: ListProps) => {
  const hasRows = props.footballPlayers && props.footballPlayers.length;

  return (
    <table className="table table-striped">
      <ListHeader></ListHeader>
      { hasRows ? <ListBody footballPlayers={props.footballPlayers}></ListBody> : <ListEmpty></ListEmpty>}
    </table>
  );
}

export default List;
