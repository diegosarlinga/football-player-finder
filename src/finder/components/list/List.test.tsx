import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import List from './List';

describe('List component tests', () => {
  let wrapper: ReactWrapper;

  describe('when rendered with elements', () => {
    beforeEach(() => {
      const props = {
        footballPlayers: [{
          contractUntil: "2020-06-30",
          dateOfBirth: "1997-02-02",
          jerseyNumber: 43,
          name: "Cameron Borthwick-Jackson",
          nationality: "England",
          position: "Left-Back",
          age: 29,
        }, {
          contractUntil: "2020-06-30",
          dateOfBirth: "1997-02-02",
          jerseyNumber: 43,
          name: "Cameron Borthwick-Jackson",
          nationality: "England",
          position: "Left-Back",
          age: 29,
        }, {
          contractUntil: "2020-06-30",
          dateOfBirth: "1997-02-02",
          jerseyNumber: 43,
          name: "Cameron Borthwick-Jackson",
          nationality: "England",
          position: "Left-Back",
          age: 29,
        }]
      };
      wrapper = mount(<List {...props}/>);
    });
  
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when rendered with no elements', () => {
    beforeEach(() => {
      const props = {
        footballPlayers: []
      };
      wrapper = mount(<List {...props}/>);
    });
  
    it('should match empty state snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});
