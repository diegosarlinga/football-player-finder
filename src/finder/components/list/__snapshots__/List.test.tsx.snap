// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component tests when rendered with elements should match snapshot 1`] = `
<List
  footballPlayers={
    Array [
      Object {
        "age": 29,
        "contractUntil": "2020-06-30",
        "dateOfBirth": "1997-02-02",
        "jerseyNumber": 43,
        "name": "Cameron Borthwick-Jackson",
        "nationality": "England",
        "position": "Left-Back",
      },
      Object {
        "age": 29,
        "contractUntil": "2020-06-30",
        "dateOfBirth": "1997-02-02",
        "jerseyNumber": 43,
        "name": "Cameron Borthwick-Jackson",
        "nationality": "England",
        "position": "Left-Back",
      },
      Object {
        "age": 29,
        "contractUntil": "2020-06-30",
        "dateOfBirth": "1997-02-02",
        "jerseyNumber": 43,
        "name": "Cameron Borthwick-Jackson",
        "nationality": "England",
        "position": "Left-Back",
      },
    ]
  }
>
  <table
    className="table table-striped"
  >
    <ListHeader>
      <thead>
        <tr>
          <th
            scope="col"
          >
            Player
          </th>
          <th
            scope="col"
          >
            Position
          </th>
          <th
            className="d-none d-sm-table-cell"
            scope="col"
          >
            Nationality
          </th>
          <th
            scope="col"
          >
            Age
          </th>
        </tr>
      </thead>
    </ListHeader>
    <ListBody
      footballPlayers={
        Array [
          Object {
            "age": 29,
            "contractUntil": "2020-06-30",
            "dateOfBirth": "1997-02-02",
            "jerseyNumber": 43,
            "name": "Cameron Borthwick-Jackson",
            "nationality": "England",
            "position": "Left-Back",
          },
          Object {
            "age": 29,
            "contractUntil": "2020-06-30",
            "dateOfBirth": "1997-02-02",
            "jerseyNumber": 43,
            "name": "Cameron Borthwick-Jackson",
            "nationality": "England",
            "position": "Left-Back",
          },
          Object {
            "age": 29,
            "contractUntil": "2020-06-30",
            "dateOfBirth": "1997-02-02",
            "jerseyNumber": 43,
            "name": "Cameron Borthwick-Jackson",
            "nationality": "England",
            "position": "Left-Back",
          },
        ]
      }
    >
      <tbody>
        <ListRow
          age={29}
          contractUntil="2020-06-30"
          dateOfBirth="1997-02-02"
          jerseyNumber={43}
          key="0"
          name="Cameron Borthwick-Jackson"
          nationality="England"
          position="Left-Back"
        >
          <tr>
            <td
              key="1"
            >
              Cameron Borthwick-Jackson
              <br
                className="d-sm-none"
              />
              <span
                className="d-sm-none text-secondary"
              >
                (England)
              </span>
            </td>
            <td
              key="2"
            >
              Left-Back
            </td>
            <td
              className="d-none d-sm-table-cell"
              key="3"
            >
              England
            </td>
            <td
              key="4"
            >
              29
            </td>
          </tr>
        </ListRow>
        <ListRow
          age={29}
          contractUntil="2020-06-30"
          dateOfBirth="1997-02-02"
          jerseyNumber={43}
          key="1"
          name="Cameron Borthwick-Jackson"
          nationality="England"
          position="Left-Back"
        >
          <tr>
            <td
              key="1"
            >
              Cameron Borthwick-Jackson
              <br
                className="d-sm-none"
              />
              <span
                className="d-sm-none text-secondary"
              >
                (England)
              </span>
            </td>
            <td
              key="2"
            >
              Left-Back
            </td>
            <td
              className="d-none d-sm-table-cell"
              key="3"
            >
              England
            </td>
            <td
              key="4"
            >
              29
            </td>
          </tr>
        </ListRow>
        <ListRow
          age={29}
          contractUntil="2020-06-30"
          dateOfBirth="1997-02-02"
          jerseyNumber={43}
          key="2"
          name="Cameron Borthwick-Jackson"
          nationality="England"
          position="Left-Back"
        >
          <tr>
            <td
              key="1"
            >
              Cameron Borthwick-Jackson
              <br
                className="d-sm-none"
              />
              <span
                className="d-sm-none text-secondary"
              >
                (England)
              </span>
            </td>
            <td
              key="2"
            >
              Left-Back
            </td>
            <td
              className="d-none d-sm-table-cell"
              key="3"
            >
              England
            </td>
            <td
              key="4"
            >
              29
            </td>
          </tr>
        </ListRow>
      </tbody>
    </ListBody>
  </table>
</List>
`;

exports[`List component tests when rendered with no elements should match empty state snapshot 1`] = `
<List
  footballPlayers={Array []}
>
  <table
    className="table table-striped"
  >
    <ListHeader>
      <thead>
        <tr>
          <th
            scope="col"
          >
            Player
          </th>
          <th
            scope="col"
          >
            Position
          </th>
          <th
            className="d-none d-sm-table-cell"
            scope="col"
          >
            Nationality
          </th>
          <th
            scope="col"
          >
            Age
          </th>
        </tr>
      </thead>
    </ListHeader>
    <ListEmpty>
      <caption
        className="text-center"
      >
        No data found for the given filters
      </caption>
    </ListEmpty>
  </table>
</List>
`;
